src/common/settings.ts

```diff
@@ -332,6 +333,56 @@ function getPreferredGlobalSetting<T>(
   return newSettings?.defaultValue;
 }
 
+/**
+ * Check if the user have configured `notebook.codeActionsOnSave` with non-notebook prefixed code actions.
+ */
+export function checkNotebookCodeActionsOnSave(serverId: string) {
+  getWorkspaceFolders().forEach((workspace) => {
+    let codeActionsOnSave: string[] = (() => {
+      const value = getConfiguration("notebook", workspace.uri).get<string[] | object>(
+        "codeActionsOnSave",
+        [],
+      );
+      if (typeof value === "object") {
+        return Object.keys(value);
+      }
+      return value;
+    })();
+
+    const genericCodeActions = codeActionsOnSave.filter(
+      (action) => action === "source.organizeImports" || action === "source.fixAll",
+    );
+
+    const ruffCodeActions = codeActionsOnSave.filter(
+      (action) =>
+        action === `source.organizeImports.${serverId}` || action === `source.fixAll.${serverId}`,
+    );
+
+    if (genericCodeActions.length > 0) {
+      // This is at info level because other extensions might be using these code actions but we still want to inform the user.
+      logger.info(
+        `The following code actions in 'notebook.codeActionsOnSave' could lead to unexpected behavior: ${JSON.stringify(
+          genericCodeActions,
+        )}. Consider using ${JSON.stringify(
+          genericCodeActions.map((action) => `notebook.${action}`),
+        )} instead.`,
+      );
+    }
```

Comment:
I'm a bit unsure if we even want this but a lot of users seems to be using the ones without `.ruff` suffix so at least the logs could help them and us.

I'm also thinking of updating the README to group config to avoid a long list of code snippets which are mostly similar and it's hard to know which config should a user pickup from them.